app/javascript/controllers/transcript_controller.js

```diff
@@ -26,17 +26,13 @@ export default class extends Controller {
     const tracks = this.player.textTracks_?.tracks_
     if (!tracks) return []
 
-    return tracks.filter(track => track.kind === 'captions')
+    return tracks.filter(track => track.kind === 'captions' && track?.cues?.cues_)
```

Comment:
Filter out caption tracks without cues *early* so we don't get `null` or `undefined` values later.